src/app/fetcher/htmlOnlyFetcher.js

```diff
@@ -35,9 +35,15 @@ export default async function fetch(url) {
 
     const mimeType = response.headers.get('content-type');
 
+    const content = await (mimeType.startsWith('text/') ? response.text() : response.buffer());
+
+    if (!content) {
```

Comment:
Why moving this inside the `fetch` module rather than leaving it to the caller to decide whether they consider it a error or not?